src/dc_tools.rs

```diff
@@ -682,6 +683,33 @@ pub fn remove_subject_prefix(last_subject: &str) -> String {
         .to_string()
 }
 
+struct ByteCounter {
+    count: usize,
+}
+
+impl ByteCounter {
+    fn new() -> Self {
+        ByteCounter { count: 0 }
+    }
+}
+
+impl std::io::Write for ByteCounter {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        self.count += buf.len();
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        Ok(())
+    }
+}
+
+pub(crate) fn count_bytes(img: &DynamicImage) -> anyhow::Result<usize> {
+    let mut writer = ByteCounter::new();
+    img.write_to(&mut writer, image::ImageFormat::Jpeg)?;
```

Comment:
Is there a guarantee that image will be encoded the same way when you call it again? Maybe count the number of bytes of actually encoded images?

Reply:
> Is there a guarantee that image will be encoded the same way when you call it again?

I don't know, but in my tests it was, and it seems unlikely to me that image encoding is non-deterministic.

> Maybe count the number of bytes of actually encoded images?

Do you mean, write it to disk and then look at the file size? Or, let `count_bytes()` keep the encoded images in RAM and then write it to disk if it's small enough? In the latter case, do you think we have to make sure that the image is not kept in RAM if it's much bigger than 20KB (to avoid keeping a 20MB image in RAM, only to realize that it's bigger than 20KB)

Reply:
I would just load the whole image into RAM. In the worst case someone tries to set a huge image as avatar and gets out-of-memory and app is killed, seems unlikely. Loading a whole image into memory is ok even for phones, uncompressed images are loaded into memory each time you scroll over them in the chat.

Current encoder is very simple and deterministic, but some realistic encoders are non-deterministic due to multi-threading. And you are encoding the image twice because of this.